src/row_codec.rs

```diff
@@ -46,25 +46,31 @@ pub(crate) fn encode_row_v0(
     key_suffix: &[u8],
     value: Option<&[u8]>,
     row_attributes: &Vec<RowAttribute>,
```

Comment:
Kinda wonder if we should have this be some kind of `RowMetadata` struct that contains `Vec<RowAttribute>` and `timestamp` (and flags?). That way, when we add more attributes in the future, we don't have to keep updating the APIs all over the place...

Reply:
see https://github.com/slatedb/slatedb/pull/265#discussion_r1805745802 for the general purpose, I did specifically break it apart here for the `row_codec` because I wanted it to be very obvious every individual thing that goes into here will increase the size of the encoded row.

Also FWIW, I think some of the confusion might be coming from the fact that `RowAttribute` and `RowAttributes` are two different classes. `RowAttributes` (with an `s`) is what you are imagining for `RowMetadata` to avoid us from needing to update a million APIs each time we add a new thing. I can rename it if it makes that clearer.

Reply:
`I think some of the confusion might be coming from the fact that RowAttribute and RowAttributes`

This spun me around a bit. Maybe we should another term for `RowAttribute` (maybe `RowFeature`?) and reserve `RowAttributes` for the logical metadata contained in a row?

Reply:
😵‍💫 omg, I didn't even notice that there were singular and plural versions. 🤦 

Wow, ok, yea I think the naming needs to be tweaked.

> RowAttributes (with an s) is what you are imagining for RowMetadata

Ya. Except I was thinking we'd want to include the Vec<RowAttribute> in the RowAttributes (lolsob) as well so we know which attributes are "active" if that makes sense.

Reply:
Something about naming and computer science 😅 in retrospect I can put that naming scheme purely in the "bad decision" category. I'll update it and make it clear!

Reply:
OK, the final naming is taken from Rohan's suggestion. `RowFeatures` are the "schema" for a row codec and `RowAttributes` is a struct with the values for the features of a specific row.

Reply:
> Ya. Except I was thinking we'd want to include the Vec in the RowAttributes (lolsob) as well so we know which attributes are "active" if that makes sense.

`[RowFeature]` is per-SST, not per-row (it's more like a schema) so I don't think it makes sense to encode it with each row.

Reply:
also relevant: https://github.com/slatedb/slatedb/pull/251#issuecomment-2395511659